core/src/main/java/io/grpc/internal/AbstractManagedChannelImplBuilder.java

```diff
@@ -379,6 +385,74 @@ public T proxyDetector(@Nullable ProxyDetector proxyDetector) {
     return thisT();
   }
 
+  @Override
+  public T defaultServiceConfig(@Nullable Map<String, ?> serviceConfig) {
+    defaultServiceConfig = parseMap(serviceConfig);
+    return thisT();
+  }
+
+  @Nullable
+  private static Map<String, ?> parseMap(@Nullable Map<?, ?> map) {
```

Comment:
This doesn't look to be parsing anything. It seems to just be making an immutable copy. Name it as such?

Reply:
Renamed to `checkMapEntryTypes()`